File: general/src/main/java/com/workoutnow/general/dtos/ExperimentalExecutionForm.java
package com.workoutnow.general.dtos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.Range;
import javax.validation.constraints.NotNull;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ExperimentalExecutionForm {
    @NotNull
    @Range(max = 500, min = 10)
    private Long weight;
    @NotNull
    @Range(max = 300, min = 50)
    private Long height;
    @NotNull
    private boolean hasSportHistory;
}

File: analytics/src/main/kotlin/com/workoutnow/analytics/dto/ExperimentalExecutionForm.kt
package com.workoutnow.analytics.dto

import com.fasterxml.jackson.annotation.JsonProperty
import javax.validation.constraints.NotNull

data class ExperimentalExecutionForm (
    @JsonProperty("weight")
    @NotNull
    val weight: Long,
    @JsonProperty("height")
    @NotNull
    val height: Long,
    @NotNull
    @JsonProperty("hasSportHistory")
    val hasSportHistory: Boolean
)
